src/prefix-color-selector.ts

```diff
@@ -0,0 +1,77 @@
+import chalk from 'chalk';
+
+export class PrefixColorSelector {
+    lastColor: string;
+    autoColors: string[];
+
+    get ACCEPTABLE_CONSOLE_COLORS() {
+        // Colors picked randomly, can be amended if required
+        return (
+            [
+                chalk.cyan,
+                chalk.yellow,
+                chalk.magenta,
+                chalk.grey,
+                chalk.bgBlueBright,
+                chalk.bgMagenta,
+                chalk.magentaBright,
+                chalk.bgBlack,
+                chalk.bgWhite,
+                chalk.bgCyan,
+                chalk.bgGreen,
+                chalk.bgYellow,
+                chalk.bgRed,
+                chalk.bgGreenBright,
+                chalk.bgGrey,
+                chalk.blueBright,
+            ]
+                // Filter out duplicates
+                .filter((chalkColor, index, arr) => {
+                    return arr.indexOf(chalkColor) === index;
+                })
+                .map(chalkColor => chalkColor.bold.toString())
+        );
+    }
+
+    constructor(private readonly prefixColors?: string[], private readonly color?: boolean) {}
+
+    getNextColor(index?: number) {
+        const cannotSelectColor = !this.prefixColors?.length && !this.color;
+        if (cannotSelectColor) {
+            return '';
+        }
+
+        const userDefinedColorForCurrentCommand =
+            this.prefixColors && typeof index === 'number' && this.prefixColors[index];
```

Comment:
nitpick: this class is always invoked from within concurrently, so it's safe to assume typescript is having our back (assuming you apply the other suggestion)

```suggestion
            this.prefixColors && this.prefixColors[index];
```

Reply:
I agree, not sure what I was thinking here 😂

Reply:
Actually, with the current implementation, the method is called without an index 
[line 69](https://github.com/open-cli-tools/concurrently/pull/296/files#diff-79dc04acf6a15308fee9ffe81596ff6718492b006c7ecec874448c9f64468e4aR69).

Reply:
implementation changed, also its great the code base is in TS now 👍🏾! when I did this initially it was still in JS